[Doc] Update Readme (#1439)
open-compass/opencompass: README.md

### 💻 Environment Setup

We highly recommend using conda to manage your python environment.

- #### Create your virtual environment

  ```bash
  conda create --name opencompass python=3.10 -y
  conda activate opencompass
  ```

> - #### Install Opencompass via pip

Reviewer: ```suggestion
- #### Install OpenCompass via pip
```

Author: Updated

  ```bash
    pip install -U opencompass

    ## Full installation (with support for more datasets)
    # pip install "opencompass[full]"

    ## Environment with model acceleration frameworks
    ## Manage different acceleration frameworks using virtual environments
    ## since they usually have dependency conflicts with each other.
    # pip install "opencompass[lmdeploy]"
    # pip install "opencompass[vllm]"

    ## API evaluation (i.e. Openai, Qwen)
    # pip install "opencompass[api]"
  ```

> - #### Install Opencompass from source

Reviewer: ```suggestion
- #### Install OpenCompass from source
```

Author: Updated

>   if you want to use opencompass's latest feature, you can also build it from source

Reviewer: ```suggestion
  if you want to use opencompass's latest features, or develop new features, you can also build it from source
```

## 🏗️ ️Evaluation

> After ensuring that OpenCompass is installed correctly according to the above steps and the datasets are prepared. Now you can start your first evaluation using Opencompass!

Reviewer: ```suggestion
After ensuring that OpenCompass is installed correctly according to the above steps and the datasets are prepared. Now you can start your first evaluation using OpenCompass!
```

> - Your first evaluation with Opencompass!

Reviewer: ```suggestion
- Your first evaluation with OpenCompass!
```

>   Opencompass support setting your configs via CLI or a python script. For simple evaluation settings we recommend using CLI, for more complex evaluation, it is suggested using the script way. You can find more example scripts under the example_scripts folder.

Reviewer: ```suggestion
  OpenCompass support setting your configs via CLI or a python script. For simple evaluation settings we recommend using CLI, for more complex evaluation, it is suggested using the script way. You can find more example scripts under the example_scripts folder.
```

  ```bash
  # CLI

>   opencompass --models hf_internlm2_chat_1_8b_sft --datasets demo_gsm8k_chat_gen

Reviewer: Please use internlm2.5-1.8b-chat for example

  # Python scripts
  opencompass example_scripts/eval_chat_demo.py
  ```

  You can find more script examples under [configs](./configs) folder.

- API evaluation

  OpenCompass, by its design, does not really discriminate between open-source models and API models. You can evaluate both model types in the same way or even in one settings.

  ```bash
  export OPENAI_API_KEY="YOUR_OPEN_API_KEY"
  # CLI
  opencompass --models gpt_4o_2024_05_13 --datasets demo_gsm8k_chat_gen

  # Python scripts
  opencompass example_scripts/eval_api_demo.py
  ```

> - Accelerated evaluation

Reviewer: ```suggestion
- Accelerated Evaluation
```

  Additionally, if you want to use an inference backend other than HuggingFace for accelerated evaluation, such as LMDeploy or vLLM, you can do so with the command below. Please ensure that you have installed the necessary packages for the chosen backend and that your model supports accelerated inference with it. For more information, see the documentation on inference acceleration backends [here](docs/en/advanced_guides/accelerator_intro.md). Below is an example using LMDeploy:

  ```bash
  # CLI

>   opencompass --models hf_internlm2_chat_1_8b_sft --datasets demo_gsm8k_chat_gen -a lmdeploy

Reviewer: Please use internlm2.5 also

  # Python scripts
  opencompass ./configs/eval_lmdeploy_demo.py
  ```

- Supported Models

  OpenCompass has predefined configurations for many models and datasets. You can list all available model and dataset configurations using the [tools](./docs/en/tools.md#list-configs).

  ```bash
  # List all configurations
  python tools/list_configs.py
  # List all configurations related to llama and mmlu
  python tools/list_configs.py llama mmlu
  ```

>   If the model is not on the list but supported by Huggingface AutoModel class, you can also evaluate it with Opencompass. You are welcome to contribute to the maintenance of the Opencompass supported model and dataset lists.

Reviewer: ```suggestion
  If the model is not on the list but supported by Huggingface AutoModel class, you can also evaluate it with OpenCompass. You are welcome to contribute to the maintenance of the Opencompass supported model and dataset lists.
```
